src/main/java/com/intuit/graphql/orchestrator/federation/EntityDataFetcher.java

```diff
@@ -85,10 +86,22 @@ public CompletableFuture<Object> get(final DataFetchingEnvironment dataFetchingE
             result -> {
               // TODO transformer
               Map<String, Object> data = (Map<String, Object>) result.get(DATA_RESPONSE_FIELD);
-              List<Map<String, Object>> _entities =
-                  (List<Map<String, Object>>) data.get(_ENTITIES_FIELD_NAME);
+              List<Map<String, Object>> _entities = (data != null) ? (List<Map<String, Object>>) data.get(_ENTITIES_FIELD_NAME) : null;
```

Comment:
Is this null check needed? wouldn't the previous line result in null pointer while casting to Map  if data is null?

Reply:
Nope; that would just remain as null. I will do the error check first, so it doesn't do an unnecessary work.